lib/assert.js

```diff
@@ -293,8 +294,16 @@ function getErrMessage(message, fn) {
   ErrorCaptureStackTrace(err, fn);
   if (errorStackTraceLimitIsWritable) Error.stackTraceLimit = tmpLimit;
 
-  overrideStackTrace.set(err, (_, stack) => stack);
-  const call = err.stack[0];
+  let call;
+  if (getEmbedderOptions().hasPrepareStackTraceCallback) {
+    overrideStackTrace.set(err, (_, stack) => stack);
+    call = err.stack[0];
+  } else {
+    const tmpPrepare = Error.prepareStackTrace;
```

Comment:
hmm actually I think both blocks are assuming `env->prepare_stack_trace_callback()` would be called here?  `Error.prepareStackTrace` doesn't even do anything if `env->prepare_stack_trace_callback()` is not configured (it's a polyfill for that hook).

Reply:
This should work: https://github.com/v8/v8/blob/main/src/execution/messages.cc#L325-L335 What we polyfill in nodejs is that `Error.prepareStackTrace` in the main realm will affect stack traces in any other realm.

Reply:
Ah didn't realize that V8 has not removed it yet..in that case I wonder if instead of using the weakmap, we can just switch to storing a `v8::Private::ForApi()` to the error to indicate that the prepare stack trace callback should use the function stored with that symbol to override the stack trace? Then the embedder can also communicate with the Node.js internals using `v8::Private::ForApi()` without requiring access to the `overrideStackTrace` map - `overrideStackTrace` isn't only used here, even.